### PR TITLE
Improve Lifetime Measurement

### PR DESCRIPTION
**Pull request by murilobalves**

### CODE REVIEW

File: apsuite/commisslib/meas_touschek_lifetime.py [hunk 1]
```diff
@@ -29,6 +29,9 @@ def __init__(self):
         super().__init__()
         self.total_duration = 0  # [s] 0 means infinity
         self.save_partial = True
+        self.save_each_nrmeas = 100
```
Comment (fernandohds564) [inline apsuite/commisslib/meas_touschek_lifetime.py:32]: I think you need to add these new attributes to the `__str__` method.

File: apsuite/commisslib/meas_touschek_lifetime.py [hunk 2]
```diff
@@ -77,17 +80,23 @@ class MeasTouschekLifetime(_BaseClass):
     RFFEAttSB = 30  # [dB] Singlebunch Attenuation
     FILTER_OUTLIER = 0.2  # Relative error data/fitting
 
-    # # calibration curves measured during machine shift in 2021/09/21:
+    # # calibration curves measured during machine studies shift in 2021/09/21:
```
Comment (fernandohds564) [inline apsuite/commisslib/meas_touschek_lifetime.py:83]: I think we can delete these old calibration curves...

File: apsuite/commisslib/meas_touschek_lifetime.py [hunk 3]
```diff
@@ -718,9 +732,13 @@ def _do_measure(self):
...
             meas['rf_voltage'].append(rfcav.dev_cavmon.gap_voltage)
+            meas['rf_frequency'].append(rfgen.frequency)
+            meas['sync_tune'].append(bbbl.sram.spec_marker1_tune)
+            meas['sync_frequency'].append(bbbl.sram.spec_marker1_freq)
+
             meas['avg_pressure'].append(press.value)
 
-            if not idx % 100 and parms.save_partial:
+            if not idx % int(parms.save_each_nrmeas) and parms.save_partial:
                 # 5 iterations of orbit correction.
```
Comment (fernandohds564) [inline apsuite/commisslib/meas_touschek_lifetime.py:742]: theses infos seem to be useful for the user. Maybe we could think of an interface where to put this kind of information about the class usage? Maybe this is something we could do for all classes. What do you think about putting it in the class docstring, @murilobalves and @xresende ?

**Review by fernandohds564 — CHANGES_REQUESTED** [review]

File: apsuite/commisslib/meas_touschek_lifetime.py (hunk 1 above)
Comment (murilobalves) [reply apsuite/commisslib/meas_touschek_lifetime.py:32]: yes, I forgot it! thank you!

File: apsuite/commisslib/meas_touschek_lifetime.py (hunk 2 above)
Comment (murilobalves) [reply apsuite/commisslib/meas_touschek_lifetime.py]: Ok, done.

**Review by jquentino — APPROVED** [review]

File: apsuite/commisslib/meas_touschek_lifetime.py (hunk 3 above)
Comment (xresende) [reply apsuite/commisslib/meas_touschek_lifetime.py:734]: yes, I think it is a good idea to have these measurement requirements explicitly explained in the class docstring.

**Review by xresende — APPROVED** [review]

**Review by fernandohds564 — APPROVED** [review]

File: apsuite/commisslib/meas_touschek_lifetime.py (hunk 3 above)
Comment (murilobalves) [reply apsuite/commisslib/meas_touschek_lifetime.py]: Nice suggestion! I updated the class docstring, please take a look. @fernandohds564 @xresende 

I recently found out a cool tool to access classes info directly from terminal. For example, is the branch of this PR:

`from apsuite.commisslib.meas_touschek_lifetime import MeasTouschekLifetime
help(MeasTouschekLifetime)`

prints the class docstring, the class methods with corresponding docstring, the inherited methods from the classes which the present class is derived and so on...

**Review by anacso17 — APPROVED** [review]

File: apsuite/commisslib/meas_touschek_lifetime.py [hunk 4]
```diff
@@ -70,23 +78,43 @@ def __str__(self):
 
 
 class MeasTouschekLifetime(_BaseClass):
-    """."""
+    """Measurement of two single-bunches current decay with BPM sum signal.
+
+    In single-bunch mode, RFFE attenuation must be adjusted in all BPMs
+    before injection, the default value is 30dB. The BPM used to be measured
```
Comment (fernandohds564) [inline apsuite/commisslib/meas_touschek_lifetime.py:84]: `used in the measurement`

File: apsuite/commisslib/meas_touschek_lifetime.py [hunk 5]
```diff
@@ -70,23 +78,43 @@ def __str__(self):
...
+    is allowed to have an attenuation of 14dB to increase the sum signal. In
```
Comment (fernandohds564) [inline apsuite/commisslib/meas_touschek_lifetime.py:85]: `increase the sum signal, if the maximum single-bunch current is bellow XXmA`

File: apsuite/commisslib/meas_touschek_lifetime.py [hunk 6]
```diff
@@ -70,23 +78,43 @@ def __str__(self):
...
+    the case that some BPMs attenuation setting fails, a local reset of RFFE
+    module at the BPM rack must be done.
+
+    The measurement is typically performed with two single-bunches with
+    different currents: one with high current (~2mA) and one with low current
+    (~0.2mA).
+
+    If correct_orbit=True in params, SOFB must be properly configured:
```
Comment (fernandohds564) [inline apsuite/commisslib/meas_touschek_lifetime.py:93]: `properly configured by hand prior to the start of the experiment.`

**Review by fernandohds564 — COMMENTED** [review]
Very nice @murilobalves ! I only have a few comments, but for me it is ok to merge as it is now!

File: apsuite/commisslib/meas_touschek_lifetime.py (hunk 4 above)
Comment (murilobalves) [reply apsuite/commisslib/meas_touschek_lifetime.py:84]: ok

File: apsuite/commisslib/meas_touschek_lifetime.py (hunk 5 above)
Comment (murilobalves) [reply apsuite/commisslib/meas_touschek_lifetime.py]: ok, with 3.0mA and attenuation of 14dB it was observed that the sum signals are within the GCA recommendations. I added this info in the docstring as suggested.

File: apsuite/commisslib/meas_touschek_lifetime.py (hunk 6 above)
Comment (murilobalves) [reply apsuite/commisslib/meas_touschek_lifetime.py]: ok

File: apsuite/commisslib/meas_touschek_lifetime.py [hunk 7]
```diff
@@ -70,23 +78,55 @@ def __str__(self):
...
+    The measurement is typically performed with two single-bunches with
+    different currents: one with high current (~2mA) and one with low current
+    (~0.2mA).
+
+    It was observed that the calibration curves BPM Sum -> Current and also the
+    DCCT offset considerably changed during a time span of about 5 months (
+    observed between machine studies in November 2021 and April 2022).
+    Therefore, it is recommended to re-measure the calibration curves prior to
+    the start of the experiment.
+    [TO-DO]: Bring the calibration curve measurement and analysis from jupyter-
```
Comment (xresende) [inline apsuite/commisslib/meas_touschek_lifetime.py:100]: @murilo, you can use comments starting with  `# NOTE:` in order to signal to-do activities. Code understands with signalling. maybe remove this part from docstring and add as a NOTE comment...

**Review by xresende — APPROVED** [review]

File: apsuite/commisslib/meas_touschek_lifetime.py (hunk 7 above)
Comment (murilobalves) [reply apsuite/commisslib/meas_touschek_lifetime.py]: done, thanks! I was wondering how to include these to-do activities. I didn't know the TODO signalling in python, thank you!

**Pull request merged**